postreise/analyze/transmission/tests/test_utilization.py

```diff
@@ -0,0 +1,82 @@
+import pandas as pd
+from powersimdata.input.tests.test_helpers import check_dataframe_matches
+from powersimdata.tests.mock_grid import MockGrid
+
+from postreise.analyze.transmission.utilization import (
+    _count_hours_gt_threshold,
+    generate_cong_stats,
+    get_utilization,
+)
+
+mock_branch = {
+    "branch_id": [101, 102],
+    "branch_device_type": ["Line"] * 2,
+    "rateA": [20, 10],
+    "from_lat": [47, 47],
+    "from_lon": [122, 122],
+    "to_lat": [47.12, 46.33],
+    "to_lon": [122.23, 121.11],
+}
+
+mock_grid = MockGrid(grid_attrs={"branch": mock_branch})
+
+mock_pf = pd.DataFrame(
+    {
+        101: [12, -8, -18, 2],
+        102: [1, 8, 9, 6],
+    },
+    index=pd.date_range(start="2016-01-01", periods=4, freq="H"),
+)
+
+
+def test_get_utilization():
+    expected = pd.DataFrame(
+        {101: [0.6, 0.4, 0.9, 0.1], 102: [0.1, 0.8, 0.9, 0.6]}, index=mock_pf.index
+    )
+    check_dataframe_matches(expected, get_utilization(mock_grid.branch, mock_pf))
+
+
+def test_get_utilization_median():
+    expected = pd.DataFrame({101: [0.5], 102: [0.7]})
+    check_dataframe_matches(
+        expected,
+        get_utilization(mock_grid.branch, mock_pf, median=True),
+    )
+
+
+def test_count_hours_gt_threshold():
+    threshold = 0.25
+    expected = pd.Series(data=[3, 3], index=[101, 102])
+
+    utilization = get_utilization(mock_grid.branch, mock_pf)
+    assert _count_hours_gt_threshold(utilization, threshold).equals(expected)
+
+
+def test_generate_cong_stats():
+    util = [0.5, 0.75, 0.85]
+    threshold = [0.5, 0.25, 0.05]
+    n = len(mock_pf)
+
+    expected = pd.DataFrame(
+        {
+            "capacity": [20.0, 10.0],
+            # "branch_device_type": ["Line"] * 2,
```

Comment:
What is `branch_device_type` in the returned data frame?

Reply:
There is a query on it in the `utilization` module to select `Line` only, see the `generate_cong_stats` function. For this reason, the column needs to be added in the mock `branch` data frame

Reply:
Yeah, my question is why we can't keep it here (not dropping below as well) and the comparison should go through?

Reply:
Sorry, yes I can uncomment it. I did not see it was commented.

Reply:
I remember now what is going on with `branch_device_type`. The `check_dataframe_matches` fails to compare strings.

Reply:
I'm wondering whether we can use `assert_frame_equal` from `pandas.testing` instead:
```python
from pandas.testing import assert_frame_equal
...
    assert_frame_equal(expected, statistics.drop("dist", axis=1))
...
```

Reply:
Done